src/components/anonymous/ChoosePassword.js

```diff
@@ -35,11 +35,11 @@ function ChoosePassword({ match }) {
     setInputs(inputs => ({ ...inputs, [name]: value }));
   }
 
-  const checkComplexity = password => password.length >= 6;
+  const checkComplexity = new RegExp(/((?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@*#$%]).{8,199})/);
```

Comment:
peux-tu mettre en commentaire la source (URL) ?

Reply:
s'il y en a une :-D

Reply:
et pourquoi limiter à 199 ?

Reply:
:) Tu as deviné :  https://stackoverflow.com/questions/23699919/regular-expression-for-password-complexity

Reply:
j'ai testé ;)

Reply:
Pour la limite c'est lié à Mattermost : Must be at least 8 characters long and less than 200, have at least one lower char, one upper char, one digit and one special char

Reply:
il manque plusieurs "caractères spéciaux", par exemple mon générateur de mdp utilise cette liste : 

!@#$%^&*

Reply:
Ok je rajoute !

Reply:
note pour la longueur : mon générateur offre une longueur max de 128 caractères, je ne sais pas pour les autres. Mais si on n'a pas de contrainte technique pour la longueur (taille hachée / champ mongo) peut-être mettre plus ??

Reply:
si aucun générateur courant ne va au delà de 128, ça devrait aller quand même, je vous laisse décider.

Reply:
Tu préconiserais donc de remplacer 199 par 128 ?

Reply:
OK vu ensemble on laisse 199